### ISSUE TITLE
`MetricsProcessor.lr_schedulers` not assigned in `Trainer.__init__`

### ISSUE THREAD

**Issue by balancap** [pytorch/torchtitan#1677]
### Bug description

The `optimizers` field is being set, but not `lr_schedulers`: https://github.com/pytorch/torchtitan/blob/main/torchtitan/train.py#L298

### Versions

`main` branch

**Comment by wwwjn** [comment]
This is because `MetricsProcessor` class doesn't use `lr_schedulers`. While the class declares the `lr_schedulers` attribute, it's never actually used in any of the class methods

Learning rate is accessed differently: When learning rate information is needed for logging, it's accessed directly from `self.lr_schedulers.schedulers[0].get_last_lr()[0]` in the training loop (line 474 in train.py) and logged by self.metrics_processor.log(), not through the `MetricsProcessor`

**Comment by tianyu-l** [comment]
Agree with @wwwjn that we are not using it, so it is not a strict bug. There might be some historical reasons why it's defined here https://github.com/pytorch/torchtitan/blob/main/torchtitan/components/metrics.py#L323

@balancap we are ok with setting it in `train.py`, if your fork needs it. Please submit a PR.

**Comment by balancap** [comment]
Thanks, I'll submit a small PR. Indeed, it would helpful for us if it is set, as we are using a custom `TrainSpec.build_metrics_processor_fn` to retrieve additional logging information.

**Comment by discobot** [comment]
The same omission exists in the torchft `FaultTolerantTrainer.__init__`, which duplicates the metrics-processor wiring block from `Trainer.__init__` — fix for both trainers in #3644.
